float64 left_front_pos
float64 left_back_pos
float64 right_front_pos
float64 right_back_pos